src/PIL/TiffImagePlugin.py

```diff
@@ -2202,6 +2202,12 @@ def fixIFD(self) -> None:
             if tag in self.Tags:
                 cur_pos = self.f.tell()
 
+                tagname = TiffTags.lookup(tag).name
+                typname = TYPES.get(field_type, "unknown")
+                msg = f"fixIFD: {tagname} ({tag}) - type: {typname} ({field_type})"
+                msg += f"- type size: {field_size} - count: {count}"
+                logger.debug(msg)
```

Comment:
Rather than evaluating the full string every time, shall we only do so when actually debug logging? Something like:

```suggestion
                logger.debug(
                    "fixIFD: %s (%d) - type: %s (%d) - type size: %d - count: %d",
                    TiffTags.lookup(tag).name,
                    tag,
                    TYPES.get(field_type, "unknown"),
                    field_type,
                    field_size,
                    count,
                )
```
(Note: I've not checked each these are the correct format specifiers.)